source/kraken/realtime.cpp

```diff
@@ -281,6 +256,76 @@ static bool is_added_stop_time(const transit_realtime::TripUpdate& trip_update)
     return false;
 }
 
+static bool is_handleable(const transit_realtime::TripUpdate& trip_update,
+                          const nt::PT_Data& pt_data,
+                          const bool is_realtime_add_enabled,
+                          const bool is_realtime_add_trip_enabled)
+{
+    namespace bpt = boost::posix_time;
+
+    auto log = log4cplus::Logger::getInstance("realtime");
+
+    // if is_realtime_add_enabled = false, this disables the adding trip option too
+    if ((! is_realtime_add_enabled && is_added_stop_time(trip_update)) ||
+        (! is_realtime_add_enabled && is_added_trip(trip_update))) {
+        LOG4CPLUS_DEBUG(log, "Add stop time is disabled: ignoring trip update id "
+                << trip_update.trip().trip_id()
+                << " with ADDED/ADDED_FOR_DETOUR stop times or ADDITIONAL_SERVICE effect");
+        return false;
+    }
+    if (! is_realtime_add_trip_enabled && is_added_trip(trip_update)) {
+        LOG4CPLUS_DEBUG(log, "Add trip is disabled: ignoring trip update id "
+                << trip_update.trip().trip_id()
+                << " with ADDITIONAL_SERVICE effect");
+        return false;
+    }
+    // Case 1: the trip is cancelled
+    if (trip_update.trip().schedule_relationship() == transit_realtime::TripDescriptor_ScheduleRelationship_CANCELED) {
+        return true;
+    }
+    // Case 2: the trip is not cancelled, but modified (ex: the train's arrival is delayed or the train's
+    // departure is brought forward), we check the size of stop_time_update because we can't find a proper
+    // enum in gtfs-rt proto to express this idea
+    if (is_circulating(trip_update)) {
+
+        // check if company id exists
+        if (trip_update.trip().HasExtension(kirin::company_id) &&
+            pt_data.companies_map.find(trip_update.trip().GetExtension(kirin::company_id)) == pt_data.companies_map.end()) {
+            LOG4CPLUS_DEBUG(log, "Trip company id "
+                    << trip_update.trip().GetExtension(kirin::company_id)
+                    <<  " doesn't exist: ignoring trip update id "
+                    << trip_update.trip().trip_id());
+            return false;
```

Comment:
The situation I can imagine here (same for physical_mode below) is:
We run this kraken version + old kirin version (rollback for a problem or because IRE is back...).
Then Kraken would just reject all TripUpdates as they do not implem company_id or physical_mode_id, no?

So I think we should reject those only when we absolutely need it (trip add?). But it should be mostly backward compatible.

Reply:
This is probably the case for GTFS-RT already.
We have to plug GTFS-RT on this PR to check the behavior **#QA**

Reply:
Tested plugging Sherbrooke's GTFS-RT on kirin: it looks like it's working (checked on departures).
To me, it still looks dangerous to over-reject here.

Reply:
To be clear, we check the _company id_ or the _physical mode id_ only if the GTFS-RT contains the extension. 
I think Sherbrooke's GTFS-RT doesn't send the extension ?

Reply:
Issue is the same for company and physical_mode.
True, I checked that for Sherbrooke Kirin doesn't send the extension, so it works.
Still, on feeds that are not adding trips, I'm not at all confident with the idea of being stricter than we used to be.
We agreed on a solution orally:
* rejecting strictly only on trip-adds